src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -103,5 +104,18 @@ public static byte[] GenerateKeyBytes(int sizeInBits)
 
             return key;
         }
+
+        /// <summary>
+        /// Gets all decryption keys.
+        /// </summary>
+        public static IEnumerable<SecurityKey> GetAllDecryptionKeys(TokenValidationParameters validationParameters)
+        {
+            if (validationParameters.TokenDecryptionKey != null)
+                yield return validationParameters.TokenDecryptionKey;
```

Comment:
'yield' can cause an issue with the memory pressure.
It would be best if we could avoid this call.

Reply:
This has been changed so that this method returns a list of all the decryption keys instead of using yield.